Added filter to make javascript variables

ndp-opendap opened the pull request:
names safe from crazy DAP var names.

jgallagher59701 commented:
So the fix is to translate all the problem characters to underscores? Sound OK. There could be conflicts like two variables hot+ and hot- but, really the fix is complicated and it's unlikely. Is the problem any of those chars because they cannot be in a js name? 

Anyway, my vote is merge!

ndp-opendap commented:
New implementation of change for variables names. Now they are simply numbered and immune from the various name content of the DAP vars and attributes.

Additionally I found the source of one of our pesky bugs in the DAP4 IFH wherein Attribute values where borking the processing stack. The problem I fixed was in JSON-LD generation where the Attribute values must be first encoded for JSON and then encoded for HTML. Because. Encoding. Fixed now.
